plugins/gssapi.c

```diff
@@ -1792,7 +1800,7 @@ static int gssapi_client_mech_step(void *conn_context,
          * two layers (say TLS and GSSAPI) to both provide these services.
          * So if we do not suppress these flags a SASL/GSS-SPNEGO negotiation
          * over, say, LDAPS will fail against Windows Servers */
-	} else if (params->props.max_ssf == 0) {
+	} else if (params->props.max_ssf <= 1) {
```

Comment:
I don't see why is this change needed.

Reply:
Because krb5 sets both confidentiality and Integrity flags, we still need to remove the confidentiality flag from the defaults when we are asking just integrity (ssf = 1) not only when we require no security layer at all

Reply:
Oh I see, although if it doesn't solve any real problem I wouldn't bother.

Looking again at the code however, I think you can change it to simple "} else {" without any condition.

Reply:
Another simple way could be to always acquire creds and to always set GSS_KRB5_CRED_NO_CI_FLAGS_X, since that's what we want.  I know we had issues with acquiring creds but as long as we pass GSS_C_NO_NAME it should be equivalent, and we'd need it for other features.